lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -485,17 +513,20 @@ WasmBytecodeGenerator::EmitBlock()
     if (m_reader->IsBinaryReader())
     {
         UINT blockCount = m_reader->m_currentNode.block.count;
-        if (blockCount <= 0)
+        if (blockCount > 0)
         {
-            throw WasmCompilationException(_u("Invalid block node count"));
```

Comment:
Where does the validation of block node counts happen now? Perhaps it should be validated in the WasmBinaryReader.